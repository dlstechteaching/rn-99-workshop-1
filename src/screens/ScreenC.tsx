import {
	StyleSheet,
	View,
	Text,
	SectionList,
	TouchableOpacity,
	ImageBackground,
} from 'react-native';
import { useNavigation } from '@react-navigation/native';
import { dataForCScreen } from './hardcoded_data';
import { MyEntryPoint } from '../components/common/MyEntryPoint';
import { Ionicons } from '@expo/vector-icons';
import { StatusBar } from 'expo-status-bar';
import { SafeAreaView } from 'react-native-safe-area-context';
import images from '../assets';

export const ScreenC: React.VFC = () => {
	const { goBack } = useNavigation();

	const sections = dataForCScreen;

	return (
		<View style={styles.container}>
			<StatusBar hidden={true} />
			<View style={styles.navbarWrapper}>
				<SafeAreaView edges={['left', 'top', 'right']} style={styles.navbar}>
					<View style={styles.navLeft}>
						<TouchableOpacity onPress={goBack} style={styles.button}>
							<Ionicons name={'arrow-back'} size={32} color="grey" />
						</TouchableOpacity>
					</View>
					<View style={styles.navCenter}>
						<Text>Settings</Text>
					</View>
					<View style={styles.navRight} />
				</SafeAreaView>
			</View>

			<View style={styles.header}>
				<ImageBackground
					source={images.settings}
					resizeMode="contain"
					style={styles.image}
					imageStyle={{ height: '100%', width: 'auto', right: -100 }}
				>
					<View style={styles.textHeader}>
						<Text style={styles.titleHeader}>Check your profile</Text>
						<Text style={styles.subtitleHeader}>jonathansmith@gmail.com</Text>
						<TouchableOpacity onPress={goBack} style={styles.buttonHeader}>
							<Text style={{ color: 'black' }}>View</Text>
						</TouchableOpacity>
					</View>
				</ImageBackground>
			</View>

			<SectionList
				sections={sections}
				keyExtractor={(item, index) => item.title + index}
				renderItem={({ item }) => <MyEntryPoint {...item} />}
				renderSectionHeader={({ section: { title } }) => (
					<Text style={styles.sectionTitle}>{title}</Text>
				)}
			/>
		</View>
	);
};

const styles = StyleSheet.create({
	container: {
		flex: 1,
		flexDirection: 'column',
		padding: 30,
		width: '100%',
		backgroundColor: 'rgb(253, 246, 239)',
		justifyContent: 'space-between',
	},
	navbarWrapper: {
		flexDirection: 'row',
		justifyContent: 'space-between',
		alignItems: 'center',
	},
	navbar: {
		width: '100%',
		flexDirection: 'row',
		alignItems: 'center',
	},
	navLeft: {
		alignItems: 'flex-start',
		flex: 1,
	},
	navCenter: {
		alignItems: 'center',
		flex: 3,
	},
	navRight: {
		alignItems: 'flex-end',
		flex: 1,
	},
	header: {
		backgroundColor: 'white',
		marginVertical: 10,
		borderRadius: 10,
		height: '20%',
		overflow: 'hidden',
	},
	image: {
		flex: 1,
		justifyContent: 'flex-start',
		overflow: 'hidden',
	},
	textHeader: {
		paddingLeft: 30,
		paddingVertical: 15,
		flexDirection: 'column',
	},
	titleHeader: {
		fontSize: 22,
		marginBottom: 5,
	},
	subtitleHeader: {
		fontSize: 12,
		color: 'rgb(152, 152, 152)',
		marginBottom: 20,
	},
	buttonHeader: {
		height: 45,
		width: '40%',
		alignItems: 'center',
		justifyContent: 'center',
		backgroundColor: 'rgb(241, 170, 105)',
		borderRadius: 15,
	},
	sectionTitle: {
		padding: 10,
	},
	button: {
		borderRadius: 35,
		flexShrink: 1,
		backgroundColor: 'rgb(220, 205, 207)',
		height: 55,
		width: 55,
		display: 'flex',
		alignItems: 'center',
		justifyContent: 'center',
	},
});
