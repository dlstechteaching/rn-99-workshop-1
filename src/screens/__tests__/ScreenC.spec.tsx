import { useNavigation } from "@react-navigation/native";
import { shallow, ShallowWrapper } from "enzyme";
import React from "react";
import { ScreenC } from "../ScreenC";

jest.mock('@react-navigation/native');

describe("ScreenC tests", () => {
  let component: ShallowWrapper;
  const navigationMock = {
      navigate: jest.fn()
  };

  const itemMock = {
    title: 'test',
    iconName: 'iconName'
  };

  beforeEach(() => {
    (useNavigation as jest.Mock).mockReturnValue(navigationMock);
  });

  it("should render properly", () => {
    givenComponent();
    thenItRendersProperly();
  });

  it("should render section list properly", () => {
    givenComponent();
    thenItRendersListProperly();
  });

  function givenComponent() {
    component = shallow(<ScreenC />);
  }

  function thenItRendersListProperly() {

    shallow(component
      .find('SectionList')
      .props()
      // @ts-ignore
      .renderSectionHeader({
          section: {
              title: 'title'
          }
      }));
    
    component
      .find('SectionList')
      .props()
      // @ts-ignore
      .keyExtractor({ title: 't'}, 0);

    shallow(component
      .find('SectionList')
      .props()
      // @ts-ignore
      .renderItem({ item: itemMock}));
  }

  function thenItRendersProperly() {
    expect(component).toMatchSnapshot();
  }

});