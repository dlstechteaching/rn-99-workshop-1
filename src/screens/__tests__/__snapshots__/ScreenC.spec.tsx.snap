// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScreenC tests should render properly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "rgb(253, 246, 239)",
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "space-between",
      "padding": 30,
      "width": "100%",
    }
  }
>
  <ExpoStatusBar
    hidden={true}
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <RNCSafeAreaView
      edges={
        Array [
          "left",
          "top",
          "right",
        ]
      }
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "width": "100%",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "flex-start",
            "flex": 1,
          }
        }
      >
        <ForwardRef
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "rgb(220, 205, 207)",
              "borderRadius": 35,
              "display": "flex",
              "flexShrink": 1,
              "height": 55,
              "justifyContent": "center",
              "width": 55,
            }
          }
        >
          <Icon
            allowFontScaling={false}
            color="grey"
            name="arrow-back"
            size={32}
          />
        </ForwardRef>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 3,
          }
        }
      >
        <Text>
          Settings
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "flex-end",
            "flex": 1,
          }
        }
      />
    </RNCSafeAreaView>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "white",
        "borderRadius": 10,
        "height": "20%",
        "marginVertical": 10,
        "overflow": "hidden",
      }
    }
  >
    <ImageBackground
      imageStyle={
        Object {
          "height": "100%",
          "right": -100,
          "width": "auto",
        }
      }
      resizeMode="contain"
      source={1}
      style={
        Object {
          "flex": 1,
          "justifyContent": "flex-start",
          "overflow": "hidden",
        }
      }
    >
      <View
        style={
          Object {
            "flexDirection": "column",
            "paddingLeft": 30,
            "paddingVertical": 15,
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 22,
              "marginBottom": 5,
            }
          }
        >
          Check your profile
        </Text>
        <Text
          style={
            Object {
              "color": "rgb(152, 152, 152)",
              "fontSize": 12,
              "marginBottom": 20,
            }
          }
        >
          jonathansmith@gmail.com
        </Text>
        <ForwardRef
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "rgb(241, 170, 105)",
              "borderRadius": 15,
              "height": 45,
              "justifyContent": "center",
              "width": "40%",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "black",
              }
            }
          >
            View
          </Text>
        </ForwardRef>
      </View>
    </ImageBackground>
  </View>
  <SectionList
    data={Array []}
    disableVirtualization={false}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    onEndReachedThreshold={2}
    renderItem={[Function]}
    renderSectionHeader={[Function]}
    scrollEventThrottle={50}
    sections={
      Array [
        Object {
          "data": Array [
            Object {
              "desc": "Customize notifications",
              "iconName": "ios-notifications-outline",
              "title": "Notifications",
            },
            Object {
              "desc": "Customize it more to fit your usage",
              "iconName": "ellipsis-horizontal-outline",
              "title": "More customization",
            },
          ],
          "title": "General",
        },
        Object {
          "data": Array [
            Object {
              "iconName": "phone-portrait-outline",
              "title": "Contact",
            },
            Object {
              "iconName": "mail",
              "title": "Feedback",
            },
            Object {
              "iconName": "shield",
              "title": "Privacy Policy",
            },
            Object {
              "iconName": "alert-outline",
              "title": "About",
            },
          ],
          "title": "Support",
        },
      ]
    }
    stickySectionHeadersEnabled={true}
    updateCellsBatchingPeriod={50}
    windowSize={21}
  />
</View>
`;
